lib/api/completeMultipartUpload.js

```diff
@@ -302,10 +309,38 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                         storedPartsAsObjects, objMD));
         },
         function waterfall6(destinationBucket, dataLocations,
-            metaStoreParams, mpuBucket, mpuOverviewKey, aggregateETag,
-            storedPartsAsObjects, objMD, next) {
-            services.metadataStoreObject(destinationBucket.getName(), undefined,
-                dataLocations, metaStoreParams, err => {
+                            metaStoreParams, mpuBucket, mpuOverviewKey,
+                            aggregateETag, storedPartsAsObjects, objMD, next) {
+            const serverSideEncryption =
+                      destinationBucket.getServerSideEncryption();
+            let pseudoCipherBundle = null;
+            if (serverSideEncryption) {
+                pseudoCipherBundle = {
+                    algorithm: destinationBucket.getSseAlgorithm(),
+                    masterKeyId: destinationBucket.getSseMasterKeyId(),
+                };
+                for (let i = 0; i < storedPartsAsObjects.length; i++) {
+                    const part = storedPartsAsObjects[i];
+                    if (part.sseAlgorithm !== pseudoCipherBundle.algorithm) {
+                        next(errors.InvalidPart);
```

Comment:
How would we have a part with the wrong algorithm since the algorithm is on the bucket?  Is this really an internal error?